heron/statemgrs/src/java/com/twitter/heron/statemgr/FileSystemStateManager.java

```diff
@@ -182,6 +183,13 @@ public Lock getLock(String topologyName, LockName lockName) {
   }
 
   @Override
+  public ListenableFuture<TopologyMaster.MetricsCacheLocation> getMetricsCacheLocation(
```

Comment:
https://github.com/twitter/heron/blob/master/heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java#L215

Also the new locations need to be added to SchedulerStateManagerAdaptor